models/articles.models.js

```diff
@@ -0,0 +1,22 @@
+const { response } = require("../app");
+const db = require("../db/connection");
+
+exports.selectArticleById = (article_id) => {
+  return db
+    .query("SELECT * FROM articles WHERE article_id=$1", [article_id])
+    .then(({ rows }) => {
+      if (rows.length === 0) {
+        return Promise.reject({
+          status: 400,
+          msg: "Bad Request",
+        });
+      }
+      return rows;
+    })
+    .catch((err) => {
+      return Promise.reject({
+        status: 400,
+        msg: "Bad Request",
+      });
+    });
```

Comment:
this catch block should be in the controller and we shouldn't need 2 promise.rejects. We can just pass the error to next()